lib/members_page.rb

```diff
@@ -0,0 +1,70 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'scraped'
+require_relative './decorators/party_name_attributes'
+
+class MembersPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::AbsoluteUrls
+  decorator PartyNameAttributes
+
+  field :party_groupings do
+    noko.xpath('//td/table/tr/td[contains(.,"Circunscripción")]/parent::tr')
+        .map do |row|
+      fragment row => PartyGroupings
+    end
+  end
+
+  class PartyGroupings < Scraped::HTML
+    field :name do
+      party_image_node[:party_name]
+    end
+
+    field :id do
+      party_image_node[:party_id]
+    end
+
+    field :members do
+      noko.xpath('following-sibling::tr')
+          .reject { |row| row.at_xpath('./td/@colspan').to_s == '3' }
+          .slice_before { |i| i.text == noko.text }
+          .first.map do |row|
+        fragment row => MemberRow
+      end.reject(&:nil?)
```

Comment:
Requiring a `reject(&:nil?)` at the end here is a red flag, especially as you're already pre-`reject`-ing.